swssconfig/swssconfig.cpp

```diff
@@ -50,23 +73,27 @@ bool write_db_data(vector<KeyOpFieldsValuesTuple> &db_items)
         size_t pos = key.find(name_delimiter);
         if ((string::npos == pos) || ((key.size() - 1) == pos))
         {
+            flush_db_data(); // flush existing valid data to DB to keep logic the same as before
             SWSS_LOG_ERROR("Invalid formatted hash:%s\n", key.c_str());
             return false;
         }
         string table_name = key.substr(0, pos);
         string key_name = key.substr(pos + 1);
-        ProducerStateTable producer(&db, table_name);
+        ProducerStateTable &producer = get_producer_table(table_name);
 
         if (kfvOp(db_item) == SET_COMMAND)
             producer.set(key_name, kfvFieldsValues(db_item), SET_COMMAND);
         else if (kfvOp(db_item) == DEL_COMMAND)
             producer.del(key_name, DEL_COMMAND);
         else
         {
+            flush_db_data(); // flush existing valid data to DB to keep logic the same as before
             SWSS_LOG_ERROR("Invalid operation: %s\n", kfvOp(db_item).c_str());
             return false;
         }
     }
+
+    flush_db_data();
```

Comment:
>flush_db_data [](http://example.com/codeflow?start=4&length=13)

I see 3 `flush_db_data()`. Is it the same just calling once outside?

Reply:
Here, I want to keep the logic as same as before. Before the change, all data will be put into redis DB until a failure occurs. So, here I call `flush_db_data()` at 3 places (2 return false, 1 return true).

Reply:
Understand your point. The pattern here is a python `finally` like syntax to achieve cleanup if out of scope. C++ has some best practice like RAII to achieve this.

Actually in this case, if you limit `table_map` scope to this function, not as a global function. The dtor will automatically flush all pipelines.

Reply:
I suppose I understand your point. RedisPipleLine will flush all data in its dtor, so if pipeline is defined in this function, it will automatically do flush. I will make this change.